src/kex_lwe_frodo/lwe_macrify.c

```diff
@@ -73,142 +51,92 @@ int MACRIFY(oqs_kex_lwe_frodo_mul_add_as_plus_e_on_the_fly)(uint16_t *out, const
 	void *aes_key_schedule = NULL;
 	OQS_AES128_load_schedule(params->seed, &aes_key_schedule, 1);
 
-	for (i = 0; i < PARAMS_N; i++) {
+	for (int i = 0; i < PARAMS_N; i++) {
+		uint16_t a_row[PARAMS_N] = {0};
 		// go through A's rows
-		memset(a_row, 0, a_rowlen);
-		for (j = 0; j < PARAMS_N; j += PARAMS_STRIPE_STEP) {
+		for (int j = 0; j < PARAMS_N; j += PARAMS_STRIPE_STEP) {
 			// Loading values in the little-endian order!
 			a_row[j] = i;
 			a_row[j + 1] = j;
 		}
 
-		OQS_AES128_ECB_enc_sch((uint8_t *) a_row, a_rowlen, aes_key_schedule, (uint8_t *) a_row);
+		OQS_AES128_ECB_enc_sch((uint8_t *) a_row, sizeof(a_row), aes_key_schedule, (uint8_t *) a_row);
 
-		for (k = 0; k < PARAMS_NBAR; k++) {
+		for (int k = 0; k < PARAMS_NBAR; k++) {
 			uint16_t sum = 0;
-			for (j = 0; j < PARAMS_N; j++) {
+			for (int j = 0; j < PARAMS_N; j++) {
 				// matrix-vector multiplication happens here
 				sum += a_row[j] * s_transpose[k * PARAMS_N + j];
 			}
 			out[i * PARAMS_NBAR + k] += sum;
-			out[i * PARAMS_NBAR + k] %= PARAMS_Q;
+			//Equivalent to %= PARAMS_Q if PARAMS_Q is a power of 2
+			out[i * PARAMS_NBAR + k] &= PARAMS_Q - 1;
 		}
 	}
 
 	OQS_AES128_free_schedule(aes_key_schedule);
-
-	ret = 1;
-	goto cleanup;
-
-err:
-	memset(out, 0, PARAMS_NBAR * PARAMS_N * sizeof(uint16_t));
-
-cleanup:
-	if (a_row != NULL) {
-		memset(a_row, 0, a_rowlen);
-		free(a_row);
-	}
-
-	if (s_transpose != NULL) {
-		memset(s_transpose, 0, PARAMS_NBAR * PARAMS_N * sizeof(int16_t));
-		free(s_transpose);
-	}
-
-	return ret;
 }
 
+
 // Generate-and-multiply: generate A column-wise, multiply by s' on the left.
-int MACRIFY(oqs_kex_lwe_frodo_mul_add_sa_plus_e_on_the_fly)(uint16_t *out, const uint16_t *s, const uint16_t *e, struct oqs_kex_lwe_frodo_params *params) {
+void MACRIFY(oqs_kex_lwe_frodo_mul_add_sa_plus_e_on_the_fly)(uint16_t *out, const uint16_t *s, const uint16_t *e, struct oqs_kex_lwe_frodo_params *params) {
 	// a (N x N)
 	// s',e' (N_BAR x N)
 	// out = s'a + e' (N_BAR x N)
 
-	int i, j, k, kk;
-	int ret = 0;
-	uint16_t *a_cols = NULL;
-	uint16_t *a_cols_t = NULL;
-
-	for (i = 0; i < PARAMS_NBAR; i++) {
-		for (j = 0; j < PARAMS_N; j++) {
-			out[i * PARAMS_N + j] = e[i * PARAMS_N + j];
-		}
-	}
-
```

Comment:
e is copied into on line 87:

`memcpy(out, e, PARAMS_NBAR * PARAMS_N * sizeof(uint16_t));`